stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/list/MessageItem.kt

```diff
@@ -384,6 +385,40 @@ internal fun DefaultMessageItemCenterContent(
 ) {
     val (message, position, _, ownsMessage, _) = messageItem
 
+    val modifier = Modifier.widthIn(max = 250.dp)
+    val isFailed = ownsMessage && message.syncStatus == SyncStatus.FAILED_PERMANENTLY
+
+    if (message.isEmojiOnly()) {
+        if (!isFailed) {
```

Comment:
Maybe we could split this into an internal small component, like:

```
@Composable
internal fun EmojiMessageContent() {
        if (!isFailed) {
            MessageContent(
                message = message,
                onLongItemClick = onLongItemClick,
                onGiphyActionClick = onGiphyActionClick,
                onImagePreviewResult = onImagePreviewResult
            )
        } else {
            Box(modifier = modifier) {
                MessageContent(
                    message = message,
                    onLongItemClick = onLongItemClick,
                    onGiphyActionClick = onGiphyActionClick,
                    onImagePreviewResult = onImagePreviewResult
                )

                Icon(
                    modifier = Modifier
                        .size(24.dp)
                        .align(BottomEnd),
                    painter = painterResource(id = R.drawable.stream_compose_ic_error),
                    contentDescription = null,
                    tint = ChatTheme.colors.errorAccent,
                )
            }
        }
}
```

Same for the other case that renders a regular message:

```
@Composable
internal fun RegularMessageContent() {

    val messageBubbleShape = when (position) {
        Top, Middle -> RoundedCornerShape(16.dp)
        else -> {
            if (ownsMessage) ChatTheme.shapes.myMessageBubble else ChatTheme.shapes.otherMessageBubble
        }
    }

    val messageBubbleColor = when {
        message.isGiphyEphemeral() -> ChatTheme.colors.giphyMessageBackground
        message.isDeleted() -> ChatTheme.colors.deletedMessagesBackground
        ownsMessage -> ChatTheme.colors.ownMessagesBackground
        else -> ChatTheme.colors.otherMessagesBackground
    }

    if (!isFailed) {
        MessageBubble(
            modifier = modifier,
            shape = messageBubbleShape,
            color = messageBubbleColor,
            border = if (messageItem.isMine) null else BorderStroke(1.dp, ChatTheme.colors.borders),
            content = {
                MessageContent(
                    message = message,
                    onLongItemClick = onLongItemClick,
                    onGiphyActionClick = onGiphyActionClick,
                    onImagePreviewResult = onImagePreviewResult
                )
            }
        )
    } else {
        Box(modifier = modifier) {
            MessageBubble(
                modifier = Modifier.padding(end = 12.dp),
                shape = messageBubbleShape,
                color = messageBubbleColor,
                content = {
                    MessageContent(
                        message = message,
                        onLongItemClick = onLongItemClick,
                        onGiphyActionClick = onGiphyActionClick,
                        onImagePreviewResult = onImagePreviewResult
                    )
                }
            )

            Icon(
                modifier = Modifier
                    .size(24.dp)
                    .align(BottomEnd),
                painter = painterResource(id = R.drawable.stream_compose_ic_error),
                contentDescription = null,
                tint = ChatTheme.colors.errorAccent,
            )
        }
    }
}
```

And then we'd have something simple here:

```
if (message.isEmojiOnly()) {
  EmojiMessageContent()
} else {
  RegularMessageContent()
}
```